api/features/versioning/models.py

```diff
@@ -163,3 +167,210 @@ def clone_to_environment(
 
         _clone.save()
         return _clone
+
+
+class VersionChangeSet(LifecycleModelMixin, SoftDeleteObject):
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    published_at = models.DateTimeField(blank=True, null=True)
+    published_by = models.ForeignKey(
```

Comment:
How is that going to work if the change is made by `master_api_key` or is that not allowed?

Reply:
Yeah, that's a good question. I did have a TODO in the code to handle API keys BUT the reason that I left it out in the end is because the only way to publish a version change set currently is through a Change Request and, at the moment, change requests themselves don't support API keys. 

I've just created an issue for this [here](https://github.com/Flagsmith/flagsmith-workflows/issues/42) but I do think that the use cases for using an API key in change request workflows are pretty slim.